packages/jest-koa-mocks/src/create-mock-context/create-mock-context.ts

```diff
@@ -70,17 +69,14 @@ export default function createContext<
     state,
   };
 
-  const protocolFallback = encrypted ? 'https' : 'http';
-  const urlObject = new URL(url, `${protocolFallback}://${host}`);
-
   const req = httpMocks.createRequest({
-    url: urlObject.toString(),
+    url,
```

Comment:
with this change handling fully-qualified URLs is broken.
```javascript
const context = createContext({ url: 'http://mystore.com/admin' });
console.log(context.host); // 'test.com'
```

What was wrong with using `url.URL` anyway? It works just fine with both fully and non-fully qualified URLs, falling back to `base` provided in its second argument 🤔

But if you think it's better how it is now for you, this should have been released as major version v6, since it's a breaking change. Just sayin' :)

Reply:
@mateus with this change handling fully-qualified URLs is broken.
```javascript
const context = createContext({ url: 'http://mystore.com/admin' });
console.log(context.host); // 'test.com'
```

What was wrong with using `url.URL` anyway? It works just fine with both fully and non-fully qualified URLs, falling back to `base` provided in its second argument 🤔

But if you think it's better how it is now for you, this should have been released as major version v6, since it's a breaking change. Just sayin' :)